targets/csharp/source/source/PlayFabGameserverModels.cs

```diff
@@ -0,0 +1,182 @@
+        public string BuildId { get; set; }
+        public string Region { get; set; }
+        public IDictionary<string, string> BuildMetadata { get; set; }
+        public IDictionary<string, string> GamePorts { get; set; }
+
+        public virtual bool ShouldLog() { return true; }
+
+        protected const string HEARTBEAT_ENDPOINT_ENV_VAR = "HEARTBEAT_ENDPOINT";
+        protected const string SERVER_ID_ENV_VAR = "SESSION_HOST_ID";
+        protected const string LOG_FOLDER_ENV_VAR = "GSDK_LOG_FOLDER";
+        protected const string TITLE_ID_ENV_VAR = "PF_TITLE_ID";
+        protected const string BUILD_ID_ENV_VAR = "PF_BUILD_ID";
+        protected const string REGION_ENV_VAR = "PF_REGION";
+        protected const string SHARED_CONTENT_FOLDER_ENV_VAR = "SHARED_CONTENT_FOLDER";
+
+        public GameserverConfiguration()
+        {
+            GameCertificates = new Dictionary<string, string>();
+            BuildMetadata = new Dictionary<string, string>();
+            GamePorts = new Dictionary<string, string>();
+        }
+    }
+
+    class EnvironmentVariableConfiguration : GameserverConfiguration
+    {
+        public EnvironmentVariableConfiguration() : base()
+        {
+            HeartbeatEndpoint = Environment.GetEnvironmentVariable(HEARTBEAT_ENDPOINT_ENV_VAR);
+            ServerId = Environment.GetEnvironmentVariable(SERVER_ID_ENV_VAR);
+            LogFolder = Environment.GetEnvironmentVariable(LOG_FOLDER_ENV_VAR);
+            SharedContentFolder = Environment.GetEnvironmentVariable(SHARED_CONTENT_FOLDER_ENV_VAR);
+            if (string.IsNullOrWhiteSpace(HeartbeatEndpoint) || string.IsNullOrWhiteSpace(ServerId))
+            {
+                throw new GSDKInitializationException("Heartbeat endpoint and Server id are required configuration values.");
+            }
+        }
+    }
+
+    class JsonFileConfiguration : GameserverConfiguration
+    {
+        private bool _shouldLog;
+
+        public override bool ShouldLog()
+        {
+            return _shouldLog;
+        }
+
+        public JsonFileConfiguration(ISerializerPlugin jsonSerializer, string fileName) : base()
+        {
+            try
+            {
+                using (StreamReader reader = File.OpenText(fileName))
+                {
+                    JsonGsdkSchema config = jsonSerializer.DeserializeObject<JsonGsdkSchema>(reader.ReadToEnd());
+
+                    HeartbeatEndpoint = config.HeartbeatEndpoint;
+                    ServerId = config.SessionHostId;
+                    LogFolder = config.LogFolder;
+                    SharedContentFolder = config.SharedContentFolder;
+                    CertificateFolder = config.CertificateFolder;
+                    GameCertificates = config.GameCertificates ?? new Dictionary<string, string>();
+                    GamePorts = config.GamePorts ?? new Dictionary<string, string>();
+                    BuildMetadata = config.BuildMetadata ?? new Dictionary<string, string>();
```

Comment:
>BuildMetadata [](start = 20, length = 13)

We should add a task to simplify this, we don't need 4 classes here - 3 configuration and one schema class. this and the schema class seem like they can be combined?